lab/skel_graph/astar/main.cpp

```diff
@@ -0,0 +1,216 @@
+// Copyright Radu Nichita 2020 radunichita99@gmail.com
+
+
+// Algoritmul A* - complexitate O(b ** d), b = branching factor, d = depth.
+
+#include <bits/stdc++.h>
+
+using Matrix = std::vector<std::vector<int>>;
+using Parents = std::vector<std::vector<std::pair<int, int>>>;
+
+class Task {
+    int n, m;
+    std::vector<int> path;
+    std::pair<int, int> source;
```

Comment:
We should face a type `Cell` as alias for `std::pair<int, int>`. I think the code would be more readable.